src/link.cpp

```diff
@@ -2713,7 +2714,15 @@ static void construct_linker_job_macho(LinkJob *lj) {
     lj->args.append("-arch");
     lj->args.append(get_darwin_arch_string(g->zig_target));
 
-    if (g->zig_target->glibc_or_darwin_version != nullptr) {
+    if (g->zig_target->glibc_or_darwin_version == nullptr) {
+        // running `zig0`; `-pie` requires versions >= 10.5; select 10.11 (5 years back)
```

Comment:
Why 10.11 instead of 10.13 (which is the minimum version supported by zig in general)